src/librustc_mir/borrow_check/type_check/mod.rs

```diff
@@ -1983,7 +1983,7 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
             }
 
             Rvalue::Repeat(operand, len) => {
-                if *len > 1 {
+                if len.try_eval_usize(tcx, self.param_env).map_or(false, |len| len > 1) {
```

Comment:
This one I'm not sure about. In case we can't evaluate the constant, should we be going into the if body, because it may not be length zero?

Reply:
Yes, we should check that the element type is `Copy` unless we know that the length is 0 or 1.

Reply:
So `false` should be replaced with `true`, or maybe you can do something like this instead:
```rust
let at_most_one_element =
    len.try_eval_usize(tcx, self.param_env).map_or(false, |len| len <= 1);
if !at_most_one_element {
```